torch/quantization/ns/graph_matcher.py

```diff
@@ -0,0 +1,315 @@
+import operator
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.nn.quantized as nnq
+import torch.nn.qat as nnqat
+import torch.nn.intrinsic.qat as nniqat
+toq = torch.ops.quantized
+
+from torch.fx import GraphModule
+from torch.fx.graph import Graph, Node
+
+from typing import Dict, Tuple, List, Optional, Set, Callable, Any
+
+# TODO(before land): delete this
+def _print_node(node: Optional[Node]) -> None:
+    if node is None:
+        print(None)
+    else:
+        print(
+            node, ', target:', node.target, ', op:', node.op,
+            ', args:', node.args, ', kwargs:', node.kwargs)
+
+def _get_output_nodes(g: Graph) -> List[Node]:
+    return [n for n in g.nodes if n.op == 'output']
+
+def get_type_a_related_to_b() -> Set[Tuple[Callable, Callable]]:
+    # TODO(future PR): allow customizations
+    # TODO(future PR): reuse existing quantization mappings
+    # TODO(future PR): add the rest of modules and ops here
+    sets_of_related_ops: List[Set[Callable]] = [
+        # conv modules
+        set([
+            nn.Conv2d,
+            nnq.Conv2d,
+            nnqat.Conv2d,
+            # Note: matching weights may not work with nniqat.ConvBn2d directly
+            # leaving that as a problem for a future PR to solve.
+            nniqat.ConvBn2d,
+        ]),
+        # linear modules
+        set([
+            nn.Linear,
+            nnq.Linear,
+            nnqat.Linear,
+        ]),
+        # linear functionals
+        set([
+            F.linear,
+            toq.linear,
+        ]),
+        # add
+        set([
+            torch.add,
+            toq.add,
+            operator.add,  # x + y
+        ]),
+    ]
+
+    type_a_related_to_b: Set[Tuple[Callable, Callable]] = set()
```

Comment:
Noob question: Cant we just verify that for 2 modules, both belong to the same set. Do we need a set of tuples here?
i.e for two graphs to match, each node i in the graph should belong to the same set in the sets_of_related_ops

Reply:
The current code precomputes "x and y are in the same set" instead of doing that at inference time.  I'm flexible though, this could be done in multiple correct ways, I don't feel strongly.